core-bundle/src/Routing/ResponseContext/Factory/Provider/ResponseContextProviderInterface.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Routing\ResponseContext\Factory\Provider;
+
+use Contao\CoreBundle\Routing\ResponseContext\ResponseContextInterface;
+
+interface ResponseContextProviderInterface
+{
+    public function supports(string $responseContextClassName): bool;
+
```

Comment:
Not sure what we decided about psalm template annotations, but this would be the perfect example for a `class-string` annotation. :smile_cat: 

```suggestion
    /**
     * @template T of ResponseContextInterface
     * @psalm-param class-string<T> $responseContextClassName
     * @return T
     */
```

Reply:
As we're using it in our tools, we might as well use them in docblocks if it makes things clearer. Added in 58f3728aa3e6ff1603dc7bfd0e80ec96df4993db